### DISCUSSION TITLE
Preservation of RTP Extension Header in RTSP to WebRTC Stream via MediaMTX

### DISCUSSION THREAD

**Discussion by perara** [bluenviron/mediamtx#3569] [Questions]
### Question

Hello MediaMTX Community,

I am currently working on a project that involves streaming from a device using RTSP to MediaMTX, and then distributing the stream to clients via WebRTC. I need to ensure that the RTP extension headers are preserved throughout this process.

Could anyone confirm if MediaMTX preserves the RTP extension headers when a device streams using RTSP to MediaMTX and clients listen to the stream via WebRTC? If there are any specific configurations or settings required to maintain these headers, I would appreciate guidance on that as well.

Thank you for your assistance!

Best regards,

**Comment by aler9** [comment]
Hello, when publishing to the server with RTSP and reading from the server with RTSP, packets are routed from the publisher to readers without any change, therefore RTP extension headers should be preserved.
There's one exception though: if packets are bigger than the UDP MTU (1420 bytes), then the server starts re-muxing them into smaller packets.
